### PR TITLE
feat: Deprecate load_data_frame

### PR DESCRIPTION
**Pull request by KowerKoint**
Generic function `gokart.task.TaskOnKart.load` also supports loading `pandas.DataFrame`.

Optional arguments `required_columns` and `drop_columns` are not necessary because you can verify/extract columns after loading `DataFrame`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -43,7 +44,6 @@ fakeredis = "*"
 mypy = "*"
 types-redis = "*"
 matplotlib = "*"
-typing-extensions = "^4.11.0"
```
Comment (hiro-o918) [inline pyproject.toml:46]: keep this line since this package is optional on production group

File: poetry.lock [hunk 2]
```diff
@@ -1,42 +1,41 @@
-# This file is automatically @generated by Poetry 1.8.3 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.8.2 and should not be changed by hand.
```
Comment (kitagry) [inline poetry.lock:1]: Could you update poetry?

File: pyproject.toml [hunk 3]
```diff
@@ -30,6 +30,7 @@ APScheduler = "*"
 redis = "*"
 dill = "*"
 backoff = "^2.2.1"
+typing-extensions = "^4.11.0"
```
[the remark this answers is not in the document]
Comment (KowerKoint) [reply pyproject.toml]: Thank you.
Use built-in `warnings.deprecated` with Python >= 3.13 on https://github.com/m3dev/gokart/pull/417/commits/8f9b3cdfac4f3880c10fe9e6805e570bcdd3a4ce .

File: poetry.lock (hunk 2 above)
Comment (KowerKoint) [reply poetry.lock]: fixed on https://github.com/m3dev/gokart/pull/417/commits/ea7525b6d6a082a4e71e90e6f3a9bf03672c5538

File: gokart/task.py [hunk 4]
```diff
@@ -5,7 +5,12 @@
 import types
 from importlib import import_module
 from logging import getLogger
+import sys
 from typing import Any, Callable, Dict, Generator, Generic, Iterable, List, Optional, Set, TypeVar, Union, overload
+if sys.version_info.minor < 13:
```
Comment (kitagry) [inline gokart/task.py:10]:
```suggestion
if sys.version_info < (3, 13):
```

**Review by kitagry — APPROVED** [review]
LGTM!

**Review by hiro-o918 — APPROVED** [review]
LGTM

**Review by hirosassa — APPROVED** [review]
Looks good!

**Comment by KowerKoint** [comment]
I fixed codes which is failed on CI.
Some `ipynb` files I've not changed was failed with latest tester packages with updating by `poetry lock`, so I recreate `poetry.lock` with adding `--no-update`.

**Comment by kitagry** [comment]
Thank you for the contribution!!

**Pull request merged**